src/main/scala/dpla/ingestion3/enrichments/StringUtils.scala

```diff
@@ -226,6 +214,66 @@ object StringUtils {
       Bracket("[","]")
     )
 
+    /**
+      * Case class that represents a pair of enclosing strings that
+      * @param openChar String that indicates start of enclosure
+      * @param closeChar String that indicates end of enclosure
+      */
     case class Bracket(openChar: String, closeChar: String)
+
+
+
+    /**
+      * Format IMT, MIME, content-type filtering normalization
+      *
+      */
+    // TODO These file references should not be hard coded
+    // TODO See VocabEnforcer -- there is some duplicate functionality that could be cleaned up
+    private def formatStopwordFiles = Seq(
+      "/formats/ohio.csv",
+      "/formats/iana-imt-types.csv"
+    )
+
+    /**
+      * Reads files listed in formatStopwordFiles and builds a Set[String]
+      * from the uncommented lines in those files
+      *
+      * @return Set[String]
+      */
+    private def getFormatStopwords: Set[String] = {
+      val fileIo = new FlatFileIO()
+      formatStopwordFiles
+        .flatMap(file => fileIo.readFileAsSeq(file))
+        .filterNot(line => line.startsWith("#"))
+        .map(_.toLowerCase())
+        .toSet
+    }
+
+    /**
+      * Gets invalid format terms by reading files specified by formatStopwordFiles. Takes those
+      * terms and constructs a massive regular expression. Removes those invalid terms from the
+      * provided string as well as empty/duplicate whitespace. This filter is case insensitive.
+      *
+      */
+    val stripInvalidFormats: SingleStringEnrichment = {
+      // TODO Move this out of the method so the file aren't re-read for every record
+      // Get stop words from files and escape forward slashes
+      // in the stop words list
+      val stopWords = getFormatStopwords
+        .map(
+          // TODO is there a better way to 'auto-escape' reserved regex chars?
+          _.replace("""/""", """\/""")
+          .replace("""+""", """\+""")
+          .replace("""-""", """\-"""))
+
+      // Create on big'ol regex from those stop words that ignores
+      // case and white space.
+      val regex = "(?i)" + stopWords
+        .map(word => s"($word)")
+        .mkString("|")
+
+      // Perform the removal and trim white space
+      value.replaceAll(regex, "").trim
+    }
```

Comment:
Have you considered that instead of creating one huge regex, you could iterate through the stop words, replacing them one at a time?  That way if there was a problem or error with one of the stop words, the others would still get cleaned up.  Not sure what its impact would be on efficiency, though.  Just a thought, feel free to keep one-regex-to-rule-them-all if you think it best.

Reply:
That was kind of what I was doing before but I found more flexibility and success with regex recently but I agree that this may not be the most sustainable.

Reply:
I trust your experience and judgement - you have done a lot more with enrichments than I!